tempto-core/src/main/java/io/trino/tempto/assertions/QueryResultValueComparator.java

```diff
@@ -284,4 +282,11 @@ private static double getDoubleValue(Object object)
     {
         return ((Number) object).doubleValue();
     }
+
+    // Make compiler happy: declare boolean is returned to avoid dead code 'return false;' after each 'fail' call.
+    private static boolean fail(Object actual, Object expected) {
+        String message = format("Invalid comparison: 'actual' value is of type '%s' while 'expected' value is of type '%s'",
+                actual.getClass().getCanonicalName(), expected.getClass().getCanonicalName());
```

Comment:
Can actual or expected be null at this point?

Reply:
made it null-safe